src/navigation/ProtectiveRoot.tsx

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import { Navigate } from 'react-router-dom';
+import RoutesList from './routes';
+import { selectUser } from '../store/auth/authSlice';
+
+type ProtectedRouteProps = {
+  outlet: JSX.Element;
+};
```

Comment:
make a new directory `src/ProtectiveRoute` and put here both (types file and component file)